### ISSUE TITLE
More than 1000 expressions in a list are not allowed on Oracle.

### ISSUE THREAD

**Issue by JoeH82** [nextcloud/server#28960]
I've allready seen, that there are some issues regarding "More than 1000 expressions in a list are not allowed on Oracle." here and all of them are closed. But i've still this message in my nextcloud logs. It seems that it appears once a day, around 11:00AM.

### Steps to reproduce
Don't really know how to reproduce, but it seems to occur regulary, even when i'm not interacting with the cloud. 
Some hints. I have some external folders with a huge amount of photos linked to the cloud. And i'm also syncing some mobile phones with a huge amount of photos to the cloud with android nextcloud app. Preview generator app is running some times a day.

### Expected behaviour
No logs of "More than 1000 expressions in a list are not allowed on Oracle.". The log is guite large and makes it impossible to use the logging feature in the webinterface.

### Actual behaviour
"More than 1000 expressions in a list are not allowed on Oracle." is logged once a day.

### Server configuration

**Operating system:**
OpenMediaVault 5, Debian 10 based, Nextcloud is running inside docker with the offical latest apache image from dockerhub.

**Web server:**
apache, as provided in official docker image

**Database:**
mariaDB as docker 

**PHP version:**
7.3.29 as provided in offical nextcloud docker

**Nextcloud version:** (see Nextcloud admin page)
22.1.1

**Updated from an older Nextcloud/ownCloud or fresh install:**
Nextcloud running since more than 2 years, with regular updates to latest version

**Where did you install Nextcloud from:**
docker

**Signing status:**
<details>
<summary>Signing status</summary>
No errors have been found.
</details>

**List of activated apps:**
<details>
<summary>App list</summary>
Enabled:
  - circles: 22.1.1
  - cloud_federation_api: 1.4.0
  - contactsinteraction: 1.2.0
  - dav: 1.18.0
  - federatedfilesharing: 1.11.0
  - files: 1.16.0
  - files_external: 1.12.1
  - files_pdfviewer: 2.3.0
  - files_rightclick: 1.1.0
  - files_sharing: 1.13.2
  - files_trashbin: 1.11.0
  - files_versions: 1.14.0
  - files_videoplayer: 1.11.0
  - gpxpod: 4.2.8
  - keeweb: 0.6.6
  - logreader: 2.7.0
  - lookup_server_connector: 1.9.0
  - nextcloud_announcements: 1.11.0
  - notifications: 2.10.1
  - oauth2: 1.9.0
  - password_policy: 1.12.0
  - previewgenerator: 3.1.1
  - provisioning_api: 1.11.0
  - settings: 1.3.0
  - sharebymail: 1.11.0
  - text: 3.3.0
  - theming: 1.12.0
  - twofactor_backupcodes: 1.10.1
  - updatenotification: 1.11.0
  - viewer: 1.6.0
  - workflowengine: 2.3.1
Disabled:
  - accessibility
  - activity
  - admin_audit
  - bruteforcesettings
  - cms_pico
  - comments
  - dashboard
  - encryption
  - federation
  - files_markdown
  - firstrunwizard
  - photos
  - polls
  - privacy
  - recommendations
  - serverinfo
  - support
  - survey_client
  - systemtags
  - tasks
  - user_ldap
  - user_status
  - weather_status

</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>
{
    "system": {
        "htaccess.RewriteBase": "\/",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
          removed
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "22.1.1.2",
        "overwrite.cli.url": "http:\/\/localhost",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "maintenance": false,
        "theme": "",
        "loglevel": 2,
        "preview_max_x": "2048",
        "preview_max_y": "2048",
        "skeletondirectory": "",
        "jpeg_quality": "75",
        "overwriteprotocol": "https",
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_sendmailmode": "smtp",
        "mail_smtpauth": 1,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "tls",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "allow_local_remote_servers": true,
        "default_phone_region": "DE",
        "app_install_overwrite": [
            "polls",
            "previewgenerator",
            "keeweb"
        ]
    }
}

</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
yes, a lot, all located on /

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

[nextcloud.log](https://github.com/nextcloud/server/files/7230281/nextcloud.log)

**Comment by JoeH82** [comment]
after digging a little bit into the log, my assumption is, that this is related to preview generator app or file cache. i can see those snippet in the log:
`"Trace":[{"file":"/var/www/html/lib/private/Files/Cache/Cache.php","line":608,"function":"execute","class":"OC\\DB\\QueryBuilder\\QueryBuilder","type":"->","args":[]},{"file":"/var/www/html/lib/private/Files/Cache/Cache.php","line":552,"function":"removeChildren","class":"OC\\Files\\Cache\\Cache","type":"->","args":[{"__class__":"OC\\Files\\Cache\\CacheEntry"}]},{"file":"/var/www/html/lib/private/Files/Cache/Updater.php","line":159,"function":"remove","class":"OC\\Files\\Cache\\Cache","type":"->","args":["appdata_ocjda3jb7z4m/preview/4"]},{"file":"/var/www/html/lib/private/Files/View.php","line":329,"function":"remove","class":"OC\\Files\\Cache\\Updater"`

When listing all files in all subfolders of mentioned folder `appdata_ocjda3jb7z4m/preview/4`, the file count is more than 5000. maybe thats the problem.

**Comment by JoeH82** [comment]
Will close the issue, not seen this error for a longer time in my log.

**Comment by nicro87** [comment]
This error not disappered for me. It appears on every place where ìn()` is used without chunking the db-request.
First I had the issue with groupfolders, wat is fixed with https://github.com/nextcloud/groupfolders/pull/1974 now.
But now it appears in `lib/private/Share20/DefaultShareProvider.php` (Line: 681 or the excute is on line 687)

It's thrown in the function `getSharesInFolder` at the moment where the in() with file_caches is executed.
I think a chunk as in the mentioned pull will fix this.

Please contact me, if you need more information.

Thank you in advance,
Nico

``` json
{
  "reqId": "9NAvzkULdRraLpzdPhq5",
  "level": 3,
  "time": "2022-04-27T11:01:43+00:00",
  "remoteAddr": "127.0.0.1",
  "user": "Administrator",
  "app": "core",
  "method": "PROPFIND",
  "url": "/remote.php/dav/files/Administrator/",
  "message": "More than 1000 expressions in a list are not allowed on Oracle.",
  "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:99.0) Gecko/20100101 Firefox/99.0",
  "version": "23.0.4.1",
  "exception": {
    "Exception": "Doctrine\\DBAL\\Query\\QueryException",
    "Message": "More than 1000 expressions in a list are not allowed on Oracle.",
    "Code": 0,
    "Trace": [
      {
        "file": "/var/www/nextcloud/lib/private/Share20/DefaultShareProvider.php",
        "line": 687,
        "function": "execute",
        "class": "OC\\DB\\QueryBuilder\\QueryBuilder",
        "type": "->",
        "args": []
      },
```

**Comment by CarlSchwan** [comment]
Hi @nicro87

This should be fixed with https://github.com/nextcloud/server/pull/32185 but in the future, this is  best to create a new issue if the backtrace is not similar (in your case it's about DefaultShareProvider and the original issue about View.php)

**Comment by nicro87** [comment]
> Hi @nicro87
> 
> This should be fixed with #32185 but in the future, this is best to create a new issue if the backtrace is not similar (in your case it's about DefaultShareProvider and the original issue about View.php)

Hi @CarlSchwan 

sorry, I know that. I was honestly too lazy. but i promise to do better.
I tried your fix and it works for me. All errors are gone.

Thanks again for the quick help,
Nico
